src/configuration.cpp

```diff
@@ -0,0 +1,702 @@
+            }
+        }
+    }
+
+    Configuration::Configuration()
+    {
+        set_configurables();
+    }
+
+    Configuration& Configuration::instance()
+    {
+        static Configuration config;
+        return config;
+    }
+
+    void Configuration::set_configurables()
+    {
+        auto& ctx = Context::instance();
+
+        // Channels
+        insert(Configurable("channels", &ctx.channels)
+                   .group("Channels")
+                   .set_env_var_name("CONDA_CHANNELS")
+                   .description("Define the list of channels")
+                   .long_description(unindent(R"(
+                    The list of channels where the packages will be search for.
+                    See also 'channel_priority'.)")));
+
+        insert(Configurable("channel_alias", &ctx.channel_alias)
+                   .group("Channels")
+                   .set_env_var_name()
+                   .description("The prepended url location to associate with channel names"));
+
+        insert(Configurable("default_channels", &ctx.default_channels)
+                   .group("Channels")
+                   .set_env_var_name()
+                   .description("Default channels used")
+                   .long_description(unindent(R"(
+                    The list of channel names and/or urls used for the 'defaults'
+                    multichannel.)")));
+
+        insert(Configurable("override_channels_enabled", &ctx.override_channels_enabled)
+                   .group("Channels")
+                   .set_env_var_name()
+                   .description("Permit use of the --overide-channels command-line flag"));
+
+        // Network
+        insert(Configurable("local_repodata_ttl", &ctx.local_repodata_ttl)
+                   .group("Network")
+                   .description("Repodata time-to-live")
+                   .long_description(unindent(R"(
+                        For a value of 0, always fetch remote repodata (HTTP 304
+                        responses respected).
+                        For a value of 1, respect the HTTP Cache-Control max-age header.
+                        Any other positive integer values is the number of seconds to
+                        locally cache repodata before checking the remote server for
+                        an update.)")));
+
+        insert(Configurable("ssl_verify", &ctx.ssl_verify)
+                   .group("Network")
+                   .set_env_var_name()
+                   .description("Verify SSL certificates for HTTPS requests")
+                   .long_description(unindent(R"(
+                        'ssl_verify' can be either an empty string (regular SSL verification),
+                        the string "<false>" to indicate no SSL verification, or a path to
+                        a directory with cert files, or a cert file..)"))
+                   .set_post_build_hook(detail::ssl_verify_hook));
+
+        insert(Configurable("cacert_path", std::string(""))
```

Comment:
is the option then translated to `--cacert-path` (underscores to dashes?)

Reply:
I don't get it, yes this configurables is used to store value passed using `--cacert-path` CLI option